internal/headers.go

```diff
@@ -106,3 +117,23 @@ func NewHeaderWriter(header *commonpb.Header) HeaderWriter {
 	}
 	return &headerWriter{header: header}
 }
+
+// WithWorkflowContext returns a new DataConverter tailored to the passed Workflow context if
+// the DataConverter implements the ContextAware interface. Otherwise the DataConverter is returned
+// as-is.
+func WithWorkflowContext(ctx Context, dc converter.DataConverter) converter.DataConverter {
+	if d, ok := dc.(ContextAware); ok {
+		return d.WithWorkflowContext(ctx)
+	}
+	return dc
+}
+
+// WithContext returns a new DataConverter tailored to the passed Activity context if
```

Comment:
nit: https://github.com/temporalio/sdk-go/blob/c8ef862218b0928928d66fda749bbe75dce9df15/internal/internal_workflow_client.go#L174 uses `WithContext` with a Context for a Workflow (but not as a workflow.Context as we don't have one there). Should perhaps broaden the comment to mention it won't always be an Activity.